src/components/enterprise-user-subsidy/enterprise-offers/data/utils.js

```diff
@@ -137,16 +138,18 @@ export const isDisableCourseSearch = (
   }
 
   const hasActiveSubPlan = subscriptionPlan?.isActive && subscriptionLicense?.status === LICENSE_STATUS.ACTIVATED;
-  const activeOffers = enterpriseOffers?.filter(item => item?.isCurrent);
-
+  const hasCouponCodes = couponCodes.filter(code => !!code?.available).length > 0;
   const allocatedOrAcceptedAssignments = learnerContentAssignments.assignments
     .filter(item => [ASSIGNMENT_TYPES.ALLOCATED, ASSIGNMENT_TYPES.ACCEPTED].includes(item.state));
+  const activeOffers = enterpriseOffers?.filter(item => item?.isCurrent);
 
-  if (allocatedOrAcceptedAssignments?.length === 0) {
+  if (hasActiveSubPlan) {
     return false;
   }
-
-  if (hasActiveSubPlan) {
+  if (hasCouponCodes) {
+    return false;
+  }
+  if (allocatedOrAcceptedAssignments?.length === 0) {
     return false;
   }
```

Comment:
Is this check redundant considering the next line already checks `allocatedOrAcceptedAssignments?.length > 0` ?  Perhaps the last line of this function can be simplified to just `return activeOffers?.length > 0;`?